### ISSUE TITLE
Use Unicode string comparison for sortable identifiers

### ISSUE THREAD

**Issue by xfix** [rust-lang/rustfmt#5075]
Rust supports Unicode identifiers (https://rust-lang.github.io/rfcs/2457-non-ascii-idents.html). I think it would make sense for rustfmt to sort identifiers according to Unicode collation rather than Unicode codepoint ordering.

The following code:

```rust
use example::{e, é, f};
```

Is formatted like this:

```rust
use example::{e, f, é};
```

As `é` is later in Unicode than `f`.

I think it would make sense to for it to be ordered like `{e, é, f}`.

Different languages have different collations. It would useful for rustfmt to provide an option to configure locale with a default setting `und` (which is good enough for most languages).

**Comment by calebcartwright** [comment]
The Rust Style Guide that codifies the rules rustfmt must obey is very explicit about the sort order being asciibetical

https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/items.md#ordering-list-import
https://github.com/rust-dev-tools/fmt-rfcs/issues/24

And that's not something that we can just change, neither in rustfmt nor the style guide. In general I agree that we need something better here (even if a non-default option) but there's a lot of things that need to get sorted (no pun intended :laughing:) before we can identify the right foot forward.

This is functionally a duplicate of #3242 as the asciibetical prescriptions apply to both, and the PR that wanted to pivot to natural sorting was targeting a hypothetical breaking major release of rustfmt that's not going to happen any time soon, and may not happen outside an ecosystem wide 2.0 rust release.
